include/lbann/models/model.hpp

```diff
@@ -241,6 +241,10 @@ class model {
   /** Save the model's weight to file */
   virtual bool save_weights(persist& p);
 
+  /** Reload the model's weights from a file */
+  virtual bool reload_weights(const std::string latest,
```

Comment:
- We should pick a more descriptive name than `latest` and document the functionality.
- The implication of `latest` is too restrictive. There are plenty of times when we might want to reload weights from many epochs ago, e.g. if training has diverged.

Reply:
Reasonable point.  Latest refers to the last checkpoint and I kept the nomenclature here so that you could easily reload from the last checkpoint of saved model state.  But we may want to make it more flexible.